### DISCUSSION TITLE
Troubleshooting new Data Source feed for TSML

### DISCUSSION THREAD

**Discussion by technologynhaa** [code4recovery/12-step-meeting-list#1549] [Q&A] [answered]
Hello there, I am working with a developer to create a new JSON feed to support our meeting listings on our Area website.

The feed we have for the Meeting Guide was missing some of the fields, so we are creating a new one just for the WordPress plugin data source. 

Here is the current test feed with one meeting listing:
https://www.areafellowship.com/api/feed/P2U_fgXSot~IpY9PK.F4kdPg-thzPJ6H/8QR_HCsofi~763fsG.PVtEdk-dakxwfg

Here is a link to our staging site where we are testing:
https://nhaastg.wpenginepowered.com/meetings/?tsml-day=any

The remaining issues I have identified include:
- Time is not displaying (showing as appointment)
- Secondary fields are not porting over, like "Conference URL Notes"
- Districts are not being created in the TSML district section

We're so close, please help!

**Comment by anchovie91471** [comment]
- Time should be in the format "HH:MM"
- It looks like Conference URL Notes is misnamed. It should be conference_url_notes not conference_notes
- I'm not certain about the district issue. You do have a district field and it is formatted correctly. I'm tempted to think this will import correctly when you get the other fields corrected.

You may want to send your developer a link to the spec to give them more specific direction on the feed. https://github.com/code4recovery/spec

**Reply by gkovats** [reply]
also `days` should be `day`, will fix the Appointment issue

**Reply by joshreisner** [reply]
for `district` to work you'll need to provide a `group` name that is unique to that group

also  `types` needs to be an array of codes (see the link that @anchovie91471 shared)

**Reply by joshreisner** [reply]
it would help debugging to send the `application/json` header so it outputs as JSON instead of HTML

**Reply by technologynhaa** [reply]
Thank you everyone, I think we're almost there having made a couple of these changes now. Time is working correctly. Conference URL Notes are showing now with the correct format/name. The only thing that remains is the District information. Districts are not populating like they do when we import a CSV with the same information.

![Screenshot 2024-10-24 at 5 55 56 PM](https://github.com/user-attachments/assets/51a1537a-9366-464e-8ed8-24bfdeffa751)

Reposting links again...

Here is the current test feed with one meeting listing:
https://www.areafellowship.com/api/feed/P2U_fgXSot~IpY9PK.F4kdPg-thzPJ6H/8QR_HCsofi~763fsG.PVtEdk-dakxwfg

Here is a link to our staging site where we are testing:
https://nhaastg.wpenginepowered.com/meetings/?tsml-day=any

**Reply by technologynhaa** [reply]
@joshreisner 

> for `district` to work you'll need to provide a `group` name that is unique to that group

Can you say more about this? I think we may be talking about two separate issues, as I have run into the issue you may be referring to where groups with the same name in different districts group together. The issue I am experiencing is that the District information is not getting added to the backend of Wordpress at all (see screenshot above). 

> also `types` needs to be an array of codes (see the link that @anchovie91471 shared)

The types are working in WordPress as expected with the current format.

![Screenshot 2024-10-24 at 6 19 26 PM](https://github.com/user-attachments/assets/17ab4e0a-eaad-4516-8122-283c6ee140b3)

![Screenshot 2024-10-24 at 6 18 36 PM](https://github.com/user-attachments/assets/37c82cb5-9645-40b3-96b2-088ca7110799)

We have a separate data feed for the Meeting Guide that has types listed out in an array of meeting codes.

**Reply by joshreisner** [reply, marked as answer]
oh wow you're right, you can in fact enter `types` as a comma-separated string - this is new, i think this is a welcome side-benefit of some import refactoring that @gkovats did. i would advise using the same feed that you use for Meeting Guide though, so you avoid having inconsistencies between the meeting lists. matching on the long name of the type is generally more fragile than using the code because those long names are liable to change every so often. for example "Step Meeting" was changed to "Step Study" but the code remained the same

and yes the `district` key is only checked when there's a `group` present. if you want to disambiguate groups you might include the netsuite group number, maybe concatenate it at the end of the name?

**Reply by technologynhaa** [reply]
@joshreisner thank you! I didn't realize the group column was not displaying in the feed, it was in the original outline of fields we mapped out in our sample but did not make it into the full listing. I think that should do the trick!
